### DISCUSSION TITLE
Track openai token usage from stream_options instead of tokenizing the response server-side

### DISCUSSION THREAD

**Discussion by kilimchoi** [langfuse/langfuse#3023] [Ideas]
### Describe the feature or potential improvement

Currently stream_options is not supported. Allow devs to add stream_options and get the token usage from openai.

**Comment by sir3mat** [comment]
any updates?

**Reply by marcklingen** [reply]
Not yet, open for contributions on this!

**Comment by hassiebp** [comment]
Tracking progress here: https://github.com/langfuse/langfuse/issues/4027

**Comment by hassiebp** [comment]
This has been released in [v2.54.4](https://github.com/langfuse/langfuse-python/releases/tag/v2.53.4). Please check out [or docs on this as well](https://langfuse.com/docs/integrations/openai/python/get-started#openai-token-usage-on-streamed-responses)
